code/modules/mob/living/simple_animal/slime/life.dm

```diff
@@ -28,28 +28,31 @@
 		return
 	..()
 
-/mob/living/simple_animal/slime/proc/AIprocess()
+/mob/living/simple_animal/slime/proc/AIprocess() // TODO: Future optimization target: Queue slime processing in a subsystem.
 	AIproc = 1
 
+	var/view_tracker = 3 // Needs to be 3 to ensure the view check runs at least once
+
 	while(AIproc)
 		if(stat == DEAD || !Target || client || buckled)
 			AIproc = 0
 			return
 
 		var/slime_on_target = 0
-		if(locate(/mob/living/simple_animal/slime) in Target.buckled_mobs)
+		if(Target.buckled_mobs.len && (locate(/mob/living/simple_animal/slime) in Target.buckled_mobs))
 			slime_on_target = 1
 
-		if((Target in view(1,src)) && Target.Adjacent(src))
-			if(attack_cooldown < world.time && CanFeedon(Target) && !slime_on_target)
+		if(get_dist(Target, src) <= 1 && Target.z == src.z && attack_cooldown < world.time)
+			if(!slime_on_target && CanFeedon(Target))
 				if(!Target.client || prob(20))
 					Feedon(Target)
 					AIproc = 0
 					return
-			if(attack_cooldown < world.time && (attacked || rabid))
+			if(attacked || rabid)
 				Target.attack_slime(src)
 				attack_cooldown = world.time + attack_cooldown_time
-		else if(Target in view(7, src))
+		else if((view_tracker % 3 != 0) || (Target in view(7, src))) // Only bother to check if target is still in view every third time
```

Comment:
I'm fairly certain there's a faster alternative to `view()` for this usecase but I can't for the life of me remember the proc name.

Reply:
Might be thinking of can_see?
`(can_see(Target, src, 7)`

Reply:
`can_see()` is horribly inaccurate.